wallet/address.go

```diff
@@ -27,8 +28,10 @@ import (
 // Address represents a identifier used in a cryptocurrency.
 // It is dependent on the currency and needs to be implemented for every blockchain.
 type Address interface {
-	// Encoder should write an address that can be decoded with Backend.DecodeAddress.
-	io.Encoder
+	// BinaryMarshaler marshals the blockchain specific address to binary
+	// format (a byte array).
+	encoding.BinaryMarshaler
+
 	// Bytes should return the representation of the address as byte slice.
 	Bytes() []byte
```

Comment:
Do we still need `Bytes()` when we have `BinaryMarshaller`? Are these consistent with each other?

Reply:
`Bytes` does the same functionality as `BinaryMarshaler`, except that it does not return an error.

In 1e659f5b14be5dbf87f5c33ed12297fe41a1a02c, I tried replacing `Bytes` with `MarshalBinary` in most places. The only place where I didn't do is `persistence/keyvalue`, because, in this place the calling function cannot return errors. And Bytes() does not return error while MarshalBinary does.

In 475d09dd70617a9e879fed5c39c9001f272eaf6a, I replaced it for keyvalue package as well, by calling `panic` when `MarshalBinary` errors. To ensure address.MarshalBinary never returns an error,  we could also introduce a generic Address test that checks for it. 

What do you think about this approach ? Should we do it this way ?

Reply:
I think that's fine 👍